### ISSUE TITLE
[Bug]: Bus layouts over 16 channels don't work with VST3 plug-ins

### ISSUE THREAD

**Issue by fholzm** [juce-framework/JUCE#1179]
### Detailed steps on how to reproduce the bug

Take the Audio Plugin Example (from `examples/CMake/AudioPlugin`), replace line 9-11 in `AudioProcessor.cpp`
```
                       .withInput  ("Input",  juce::AudioChannelSet::stereo(), true)
                      #endif
                       .withOutput ("Output", juce::AudioChannelSet::stereo(), true)
```
with
```
                       .withInput  ("Input",  juce::AudioChannelSet::ambisonic(5), true)
                      #endif
                       .withOutput ("Output", juce::AudioChannelSet::ambisonic(5), true)
```
and include the layout in `bool AudioPluginAudioProcessor::isBusesLayoutSupported`. An [assertion](https://github.com/juce-framework/JUCE/blob/bbd6ccbc863edec3150e1e024e4fa3f636802596/modules/juce_audio_plugin_client/VST3/juce_VST3_Wrapper.cpp#L3050) gets triggered.

When using 4th order ambisonics, only 4 in-/output channels instead of 25 are available in Reaper.

A 7.1.6 layout seems to be working fine, I cloud also create multiple input buses with this layout exceeding 16 channels.

### What is the expected behaviour?

No assertion and the correct channel count

### Operating systems

macOS

### What versions of the operating systems?

13.0.1

### Architectures

ARM

### Plug-in formats (if applicable)

VST3

### Plug-in host applications (DAWs) (if applicable)

Reaper

### Testing on the `develop` branch

I have not tested against the `develop` branch

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by reuk** [comment]
Ambisonics orders higher than 3 are not supported by the VST3 format, as far as I can tell. The VST3 speaker arrangement header only defines up to 16 different ambisonic speakers, allowing up to 3rd-order ambisonics.

https://steinbergmedia.github.io/vst3_doc/vstinterfaces/group__speakerArrangements.html#ga1b19ef3d9cbdfb96c008f4586b186997

**Comment by fholzm** [comment]
You are completely right, I thought VST3 would support up to 5th order. However, maybe an assertion would be helpful in the VST3 wrapper? There's already one thrown if the [order exceeds 5](https://github.com/juce-framework/JUCE/blob/bbd6ccbc863edec3150e1e024e4fa3f636802596/modules/juce_audio_basics/buffers/juce_AudioChannelSet.cpp#L481)
